### ISSUE TITLE
Interop Cloud to Prod Tests Failing - cannot lock ref 'refs/heads/jenkins-docker': reference already exists

### ISSUE THREAD

**Issue by yashykt** [grpc/grpc#17625]
Seen on #17609 

```
Status: Image is up to date for grpctesting/grpc_interop_java_114301e2c5f33ba60e806b94a21f5d75fe4e2eb5:latest
real	0m18.010s
user	0m0.107s
sys	0m0.117s
+ git branch -f jenkins-docker
fatal: cannot lock ref 'refs/heads/jenkins-docker': reference already exists
```

https://source.cloud.google.com/results/invocations/071213b3-bd97-4460-947f-197fdc6737bd/targets/grpc%2Fcore%2Fpull_request%2Flinux%2Fgrpc_interop_toprod/log

Might be a transient thing caused by another issue.

**Comment by jtattermusch** [comment]
AFAIK this is caused by  multiple docker image builds running in parallel and they race when trying to create the `jenkins-docker` ref.
https://github.com/grpc/grpc/blob/97453a3d81185c76cc870e732af3c72438e620f0/tools/run_tests/dockerize/build_interop_image.sh#L94

I think `git branch -f jenkins-docker` is there to silence the warning that we are in detached mode (and thus shorten the logs).

I think this problem happens very rarely, but I might get to fixing this at some point.
